src/main/java/sirius/search/constraints/RobustQueryParser.java

```diff
@@ -34,67 +37,79 @@
+public class RobustQueryParser implements Constraint {
 
     private static final int EXPANSION_TOKEN_MIN_LENGTH = 2;
-    private String defaultField;
-    private String input;
-    private Function<String, Iterable<List<String>>> tokenizer;
-    private boolean autoexpand;
+    private final String input;
+    private final String defaultField;
+    private final Function<String, Iterable<List<String>>> tokenizer;
+    private boolean autoExpand;
+    private QueryBuilder finishedQuery;
+    private Monoflop parsed;
 
-    /**
-     * Creates a parser using the given query, defaultField and analyzer.
-     *
-     * @param defaultField the default field to search in, if no field is given.
-     * @param input        the query to parse
-     * @param tokenizer    the function to used for tokenization of the input
-     * @param autoexpand   should single token queries be auto expanded. That will put a "*" after the resulting token
-     *                     but limits the number of expansions to the top 256 terms.
+    /*
+     * Use the #on(Object, String) factory method
      */
-    public RobustQueryParser(String defaultField,
-                             String input,
-                             Function<String, Iterable<List<String>>> tokenizer,
-                             boolean autoexpand) {
-        this.defaultField = defaultField;
+    private RobustQueryParser(String input,
+                              String defaultField,
+                              Function<String, Iterable<List<String>>> tokenizer,
+                              boolean autoExpand) {
         this.input = input;
+        this.defaultField = defaultField;
         this.tokenizer = tokenizer;
-        this.autoexpand = autoexpand;
+        this.autoExpand = autoExpand;
+        this.parsed = Monoflop.create();
+        this.finishedQuery = createQuery();
     }
 
     /**
-     * Compiles and applies the query to the given one.
+     * Creates a new constraint which queriess the given field.
      *
-     * @param query               the query to enhance with the parsed result
-     * @param failForEmptyQueries determines if a query was forced, see {@link Query#forceQuery(String, String, *                            Function)}
+     * @param input      the query to parse
+     * @param field      the field to search in
+     * @param tokenizer  the function to used for tokenization of the input
+     * @param autoExpand should single token queries be auto expanded. That will put a "*" after the resulting token
+     *                   but limits the number of expansions to the top 256 terms.
+     * @return the newly created constraint
      */
-    void compileAndApply(Query<?> query, boolean failForEmptyQueries) {
-        QueryBuilder main = compile();
-        if (main != null) {
-            if (Index.LOG.isFINE()) {
-                Index.LOG.FINE("Compiled '%s' into '%s'", query, main);
+    public static RobustQueryParser on(String input,
```

Comment:
Kann man sich eig. sparen - das ist nur wenn man so builder mit on(field).eq(value) und so macht...in dem fall passts eig. so